scripts/bump-oss-version.js

```diff
@@ -23,29 +23,42 @@ const yargs = require('yargs');
 let argv = yargs.option('r', {
   alias: 'remote',
   default: 'origin',
+})
+.option('n', {
+  alias: 'nightly',
+  type: 'boolean',
+  default: false,
 }).argv;
 
-// Check we are in release branch, e.g. 0.33-stable
-let branch = exec('git symbolic-ref --short HEAD', {
-  silent: true,
-}).stdout.trim();
-
-if (branch.indexOf('-stable') === -1) {
-  echo('You must be in 0.XX-stable branch to bump a version');
-  exit(1);
-}
-
-// e.g. 0.33
-let versionMajor = branch.slice(0, branch.indexOf('-stable'));
-
-// - check that argument version matches branch
-// e.g. 0.33.1 or 0.33.0-rc4
-let version = argv._[0];
-if (!version || version.indexOf(versionMajor) !== 0) {
-  echo(
-    `You must pass a tag like 0.${versionMajor}.[X]-rc[Y] to bump a version`,
-  );
-  exit(1);
+const nightlyBuild = argv.nightly;
+
+let version, branch;
+if (nightlyBuild) {
+  const currentCommit = exec('git rev-parse HEAD', {silent: true}).stdout.trim();
+  version = `0.0.0-${currentCommit.slice(0, 9)}`;
```

Comment:
Currently the shortest a hash can safely be in the RN repo is 8 characters, so, while it will probably take a while before 9 becomes an issue, in the changelog-generator I did settle on 10 chars for display purposes and no shortening where displaying it isn’t a concern.

Just my 2¢

Reply:
I picked 9 to align with what react is using.